### ISSUE TITLE
Approximate version

### ISSUE THREAD

**Issue by alyst** [lejon/TSne.jl#7]
Just to have it in the list. [Rtnse](https://github.com/jkrijthe/Rtsne) could be run in "exact" (what TSne.jl has now) and "approximate" modes.
In "approximate" mode it optimizes the calculation of gradient in the main loop by aggregating the forces of individual points onto a grid of specified size `theta` and using SP-Tree for an efficient representation of these "voxels". It leads to huge speed gains, especially for large datasets.

**Comment by lejon** [comment]
Yes, definitely agree! This is usually called Barnes-Hut t-SNE. I implemented it for the Java version and it has been on my TODO list for a long time for Julia, but alas, I have not had time to do it for the Julia version yet :(
